src/test/java/com/facebook/genAlgo/genepool/GenePoolTest.java

```diff
@@ -0,0 +1,82 @@
+package com.facebook.genAlgo.genepool;
+
+import com.facebook.genAlgo.evaluator.Evaluator;
+import com.facebook.genAlgo.gene.Gene;
+import com.facebook.genAlgo.mutator.MutatorService;
+import com.facebook.genAlgo.utils.RandomProvider;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.provider.ValueSource;
+import org.mockito.Mock;
+
+import java.util.List;
+
+import static org.junit.jupiter.api.Assertions.*;
+import static org.mockito.Matchers.anyObject;
+import static org.mockito.Mockito.*;
+
+class GenePoolTest {
+
+    @Mock
+    RandomProvider randomProvider;
+
+    @Mock
+    MutatorService mutatorService;
+
+    @Mock
+    Evaluator evaluator;
+
+    @Test
+    @ValueSource(ints = {2, 10, 40, 55, 287})
+    public void shouldInitializeGenes(int sizeExpected) {
+        // given
+        GenePool genePool = new GenePool(randomProvider, mutatorService, evaluator, sizeExpected);
+
+        // when
+        List<Gene> poolOfGenes = genePool.getPoolOfGenes();
+
+        // then
+        Assertions.assertEquals(sizeExpected, poolOfGenes.size());
+    }
+
+    @Test
+    @ValueSource(ints = {2, 10, 30, 55, 1000})
+    public void shouldPerformMutation(int sizeExpected) {
+        // given
+        GenePool genePool = new GenePool(randomProvider, mutatorService, evaluator, sizeExpected);
+
+        // when
+        genePool.makeMutation();
+
+        // then
+        verify(mutatorService, times(sizeExpected)).mutate(anyObject());
+    }
+
+    @Test
+    @ValueSource(ints = {2, 10, 30, 55, 1000})
+    public void shouldPerformEvaluation(int sizeExpected) {
+        // given
+        GenePool genePool = new GenePool(randomProvider, mutatorService, evaluator, sizeExpected);
+
+        // when
+        genePool.evaluateFitness();
+
+        // then
+        verify(evaluator, times(sizeExpected)).setFitness(anyObject());
+    }
+
+    @Test
+    @ValueSource(ints = {2, 10, 30, 55, 1000})
+    public void shouldPerformEvolution(int sizeExpected) {
```

Comment:
I suggest to change this test to: shouldIncreaseGenerationWhenPerformEvolution and use setGeneration to number from ValueSource.
assert that generation is increased by 1 after performEvolution.